### ISSUE TITLE
change password on android

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/server#8498]
Currently we just open up our AuthenticatorActivity, which is the same as it initial login. Depending on the server version it then shows the weblogin flow or the legacy login (with prefilled username).
As told by @MorrisJobke other user backends might return a custom link to reset/change the password.
How can we integrate this into android?

**Comment by MorrisJobke** [comment]
> How can we integrate this into android?

Especially on the fallback screen. Don't know if this is mandatory that much or if we just hide it from the legacy screen.

@tobiasKaminsky Is there a plan to remove this screen in the future? And if so: when?

**Comment by tobiasKaminsky** [comment]
I would do this only for the new weblogin flow as this is already supported back to NC11.
The legacy one is really only a fallback if somehow the weblogin view is not properly shown.

**Comment by MorrisJobke** [comment]
> I would do this only for the new weblogin flow as this is already supported back to NC11.
> The legacy one is really only a fallback if somehow the weblogin view is not properly shown.

But then you don't need it, because it is only server side code and nothing that is rendered by the Android app code, right?

**Comment by tobiasKaminsky** [comment]
As said, currently we simply open up weblogin flow, where you cannot change the password.
So we would need a similar endpoint like weblogin for our plain user backend and other values for other backends.

**Comment by MorrisJobke** [comment]
> As said, currently we simply open up weblogin flow, where you cannot change the password.
> So we would need a similar endpoint like weblogin for our plain user backend and other values for other backends.

Okay - one step back: about what are we talking and what does the android client need?

**Comment by tobiasKaminsky** [comment]
We have a "change password" button.

![2018-02-22-150202](https://user-images.githubusercontent.com/5836855/36542536-59e7fefa-17e1-11e8-84eb-6acf0b961d29.png)

I assumed that I could change the current password with this button:
enter current one, enter new one -> saved.

Same as on server:
![image](https://user-images.githubusercontent.com/5836855/36542671-c0642c80-17e1-11e8-98e7-f238bfc74bbe.png)

But my expectation was wrong.
I tested it with NC11 and legacy login and with recent master.
The whole purpose of this button is to change the password with which the user authenticates.
But this is kind of useless as the app should detect if the credentials are not valid anylonger and then show the weblogin to re-authenticate.

tl;dr: do we want to have a real "change password" in android? Or do we want to say that such stuff must be done via web ui?

**Comment by rullzer** [comment]
I'd say do it via the web.

**Comment by tobiasKaminsky** [comment]
Thanks, then nothing to do on server side ;-)
